addons/base_vat/models/res_partner.py

```diff
@@ -168,27 +177,31 @@ def check_vat(self):
                 msg = partner._build_vat_error_message(country and country.code.lower() or None, partner.vat, partner_label)
                 raise ValidationError(msg)
 
-    @api.onchange('vat', 'country_id')
-    def _onchange_check_vies(self):
-        """ Check the VAT number with VIES, if enabled. Return a non-blocking warning if the check fails."""
-        if self.env.context.get('company_id'):
-            company = self.env['res.company'].browse(self.env.context['company_id'])
-        else:
-            company = self.env.company
-        if not company.vat_check_vies:
-            return
-
-        eu_countries = self.env.ref('base.europe').country_ids
-        for eu_partner_company in self.filtered(lambda partner: partner.country_id in eu_countries and partner.is_company):
+    @api.depends('vat', 'country_id')
+    def _compute_vies_valid(self):
+        """ Check the VAT number with VIES, if enabled."""
+        for partner in self:
+            if not partner.perform_vies_check:
+                continue
             # Skip checks when only one character is used. Some users like to put '/' or other as VAT to differentiate between
             # A partner for which they didn't input VAT, and the one not subject to VAT
-            if not eu_partner_company.vat or len(eu_partner_company.vat) == 1:
+            if not partner.vat or len(partner.vat) == 1:
+                partner.vies_valid = False
                 continue
-            country = eu_partner_company.country_id
-            if self._run_vies_test(eu_partner_company.vat, country) is False:
-                self.vies_failed_message = _("The VAT number %s failed the VIES VAT validation check.", eu_partner_company.vat)
-            else:
-                self.vies_failed_message = False
+            try:
+                vies_valid = check_vies(partner.vies_vat_to_check, timeout=10)
+                partner.vies_valid = vies_valid['valid']
+                continue
+            except OSError as e:
```

Comment:
OSError? For the connection to the server? :thinking:

Reply:
Yep, it's weird, but from testing, it seems to be `OSError: [Errno 101] Network is unreachable` in cases where I deliberately cause a timeout, or use the wrong address